Glimpse.ApplicationInsights/GlimpseTelemetryChannel.cs

```diff
@@ -73,21 +166,25 @@ public void Send(ITelemetry item)
                     Category = "Application Insights",
                     Message = t.SeverityLevel == null ? t.Message : t.SeverityLevel + ": " + t.Message,
                     FromFirst = timer.Point().Offset,
-                    FromLast = CalculateFromLast(timer),
+                    FromLast = this.CalculateFromLast(timer),
                     IndentLevel = 0
                 };
-
-                MessageBroker.Publish(model);
+                this.MessageBroker.Publish(model);
             }
 
-            messageBroker.Publish(item);
-        }
+            // Filter telemetry with empty instrumentation key
+            if (!item.Context.InstrumentationKey.ToString().Equals("00000000-0000-0000-0000-000000000000"))
```

Comment:
check InstrumentationKey for null. Also InstrumentationKey is already string, no need to ```ToString``` it again, right?